applications/frontend/src/store/reducers/models.ts

```diff
@@ -0,0 +1,32 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+
+import { CallApiAction } from '../../middleware/backend'
+import { Model } from '../../types/model'
+
+const initialState: Model[] = [];
+
+const modelSlice = createSlice({
+  name: 'models',
+  initialState,
+  reducers: {
+    loadModels(state, action: PayloadAction<Model[]>) {
+      return action.payload;
+    },
+  }
+});
+
+export const fetchModels = (): CallApiAction => {
+  return ({
+    type: 'models/fetchModels',
+    payload: {
+      url: '/api/models',
```

Comment:
Better leave out api details from the business logic. See comment on the middleware.

Reply:
this is the action definition of fetching the model

Reply:
done, created osbmiddleware